build/build.py

```diff
@@ -828,29 +830,25 @@ def sign(self, distDir):
         for filename in files:
             if os.path.basename(filename) in self.docs:
                 continue
-            runCmd([gpgCmd, '--yes', '-ab', os.path.join(distDir, filename)])
+            if Path(filename).suffix == ".asc":
+                continue
+            runCmd(shlex.split(
+                f"{gpgCmd} --yes -ab {os.path.join(distDir, filename)}"))
 
-    def createArtifacts(self, jarOrWar, url=None):
-        whichDir = distDir
-        distJarOrWar = "build/dist"
-        if jarOrWar == "war":
-            whichDir = distWarDir
-            distJarOrWar = "build/dist-war"
-        ensureDirExists(whichDir)
-        self.version = validatorVersion
-        runAnt(['-Ddist=' + distJarOrWar,
-                '-Dversion=' + self.version,
-                '-f', self.buildXml],
-                ('%s-artifacts' % self.artifactId))
-
-    def createBundle(self):
-        self.createArtifacts("jar")
-        print("Building %s/%s-%s-bundle.jar" %
-              (distDir, self.artifactId, self.version))
-        self.sign(distDir)
-        runAnt(['-Dversion=' + self.version,
-                '-f', self.buildXml],
-                ('%s-bundle' % self.artifactId))
+    def createMavenArtifacts(self):
+        ensureDirExists(distDir)
+        runAnt(shlex.split(f"-Dversion={self.version} -f {self.buildXml}"),
```

Comment:
Any reason not to pass/build an array for the arguments?

Reply:
>>         `runAnt(shlex.split(f"-Dversion={self.version} -f {self.buildXml}"),`
> Any reason not to pass/build an array for the arguments?

Isn’t that what `shlex.split` does? I thought it took in the input string and turned it into an array?

Reply:
I don't know. But what you pass to runant will be joined again to run the command. So that looks like useless operation. And with simply an array code is easier to read.

Reply:
Then change all other locations where ant is invoked, for consistency... It is cleaner for git history to have one line per 'logic argument'